src/main/java/io/prestosql/plugin/tiledb/TileDBRecordCursor.java

```diff
@@ -268,17 +268,16 @@ private void initializeQuery(TileDBSplit split) throws TileDBError
             }
         }
 
-        // Build subarray for query based on split
-        subArray = buildSubArray(split, columnHandles);
-        // Get max buffer sizes to build
-        HashMap<String, Pair<Long, Long>> maxSizes = this.array.maxBufferElements(subArray);
+        // Build ranges
+        setRanges(split);
+        // Get estimated buffer sizes to build
+        HashMap<String, Pair<Integer, Integer>> maxSizes = this.query.getResultEstimations();
```

Comment:
Has the query had the ranged added at this point? We can only use `getRestultEstimations` after the query has been configured with ranges (or in more general subarray too).